Src/FluentAssertions/Equivalency/EquivalencyValidationContext.cs

```diff
@@ -80,6 +80,20 @@ public bool IsCyclicReference(object expectation)
 
     public ITraceWriter TraceWriter { get; set; }
 
+    /// <summary>
+    /// This method ensures that tracing starts with a fresh state when invoked.
+    /// </summary>
+    internal void ResetTracing()
+    {
+        // SMELL: We need to ensure that if tracing is enabled using the built-in writer,
+        // we start with a fresh instance of StringBuilderTraceWriter. We can't add extend ITraceWriter
+        // as that would be a breaking change.
+        if (TraceWriter is StringBuilderTraceWriter)
```

Comment:
Should we use `if (TraceWriter?.GetType() == typeof(StringBuilderTraceWriter))` in case a user has subclassed `StringBuilderTraceWriter`? 🤔